### PR TITLE
fixes #168 #280 added unit test for Garbage collection iterator, fixed b...

### LINKED ISSUE #168: Add unit test for GarbageCollectionIterator
The grabage collection iterator needs a comprehensive unit test.  The unit test does not have to use minifluo, it could just use the Accumulo SortedMapIterator.

### PR DESCRIPTION
**Pull request by keith-turner**
...ugs

 * fixed bug where garbage collection iter was using wrong timestamp for DEL_LOCK invalidation
 * fixed bug where ACKs where not being GCed

### CODE REVIEW

File: modules/accumulo/src/test/java/io/fluo/accumulo/iterators/GarbageCollectionIteratorTest.java [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+
+    input.add("0 f q WRITE 16","11");
+    input.add("0 f q LOCK 21", "1 f q");
+    input.add("0 f q LOCK 11", "1 f q");
+    input.add("0 f q DATA 21","18");
+    input.add("0 f q DATA 11","15");
+
+    TestData output = new TestData(newGCI(input, 10));
+
+    //System.out.println(output);
```
Comment (mikewalch) [inline modules/accumulo/src/test/java/io/fluo/accumulo/iterators/GarbageCollectionIteratorTest.java]: Might want to remove this..

File: modules/accumulo/src/test/java/io/fluo/accumulo/iterators/GarbageCollectionIteratorTest.java [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+    Assert.assertEquals(expected, output);
+
+    expected = new TestData();
+    expected.add("0 f q WRITE 52","48");
+    expected.add("0 f q WRITE 45","41 TRUNCATION");
+    expected.add("0 f q DATA 48","18");
+    expected.add("0 f q DATA 41","15");
+
+    output = new TestData(newGCI(input, 46));
+    Assert.assertEquals(4, expected.data.size());
```
Comment (mikewalch) [inline modules/accumulo/src/test/java/io/fluo/accumulo/iterators/GarbageCollectionIteratorTest.java]: Is this assertion necessary given the next statement?  Or it just to help debugging?

Comment (keith-turner) [inline modules/accumulo/src/test/java/io/fluo/accumulo/iterators/GarbageCollectionIteratorTest.java]: > Is this assertion necessary given the next statement?
When I first wrote the `TestData` class it had a bug in it and was not adding anything.  I added this code to test the test code.

File: modules/accumulo/src/test/java/io/fluo/accumulo/iterators/GarbageCollectionIteratorTest.java (hunk 1 above)
Comment (keith-turner) [inline modules/accumulo/src/test/java/io/fluo/accumulo/iterators/GarbageCollectionIteratorTest.java]: fixed in cb9e54f

File: modules/accumulo/src/test/java/io/fluo/accumulo/iterators/GarbageCollectionIteratorTest.java (hunk 2 above)
Comment (mikewalch) [inline modules/accumulo/src/test/java/io/fluo/accumulo/iterators/GarbageCollectionIteratorTest.java]: sounds good

**Comment by mikewalch** [comment]
+1

**Pull request merged**